app/components/file_attachment_list/file_attachment_list.test.js

```diff
@@ -188,17 +184,50 @@ describe('FileAttachmentList', () => {
             },
         };
 
-        const wrapper = shallow(
-            <FileAttachment {...props}/>
+        shallow(
+            <FileAttachment {...props}/>,
         );
-        expect(wrapper.state('loadingFiles')).toBe(true);
         expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledTimes(1);
         expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledWith(props.postId);
         await loadFilesForPostIfNecessaryMock();
         expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledTimes(2);
         expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledWith(props.postId);
-        await loadFilesForPostIfNecessaryMock();
-        wrapper.setProps({files: baseProps.files});
-        expect(wrapper.state('loadingFiles')).toBe(false);
+    });
+
+    test('should call getFilesForGallery on props change', async () => {
+        const loadFilesForPostIfNecessaryMock = jest.fn().mockImplementationOnce(() => Promise.resolve({data: {}}));
+        const props = {
+            ...baseProps,
+            actions: {
+                loadFilesForPostIfNecessary: loadFilesForPostIfNecessaryMock,
+            },
+        };
+
+        const wrapper = shallow(
+            <FileAttachment {...props}/>,
+        );
+
+        wrapper.instance().getFilesForGallery = jest.fn().mockImplementationOnce(() => []);
+        wrapper.setProps({files: [files[0], files[1]]});
+        expect(wrapper.instance().getFilesForGallery).toHaveBeenCalled();
+    });
+
+    test('should call loadFilesForPostIfNecessary when files object empty', async () => {
+        const loadFilesForPostIfNecessaryMock = jest.fn().mockImplementationOnce(() => Promise.resolve({data: {}}));
+        const props = {
+            ...baseProps,
+            actions: {
+                loadFilesForPostIfNecessary: loadFilesForPostIfNecessaryMock,
+            },
+        };
+
+        const wrapper = shallow(
+            <FileAttachment {...props}/>,
+        );
+        wrapper.setProps({files: []});
+        expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledTimes(1);
+        expect(props.actions.loadFilesForPostIfNecessary).toHaveBeenCalledWith(props.postId);
+        wrapper.setProps({files: []});
```

Comment:
Instead of changing `files` to a new empty array, can you instead confirm that this isn't called a second time when some other prop changes? It can be something like
```suggestion
        wrapper.setProps({test: 'test'});
```